docs/config/@config.md

```diff
@@ -0,0 +1,222 @@
+B:doTheThing=true   
+``` 
+This will produce a `boolean` config value with name `doTheThing`, the defualt value `true` and the comment provided in the config file.
+
+@Name Use
+---------
+
+You should use the `@Name` annotation to give a config a user friendly name in the config file.
+
+This has 1 property:
+| Property | Type    | Default Value |
+|        -:|:-       |      :-:      |
+|     value|`String` |      N/A      | 
+
+### Example
+```java
+@Name("FE/T for the thing")
+public static int thingFE = 50; 
+```
+This will produce the following config:
+```
+I:"FE/T for the thing"=50
+```
+
+@RangeInt Use
+-------------
+
+You should use `@RangeInt` to limit an `int` or `Integer` config value. 
+
+It has a couple of properties:
+
+| Property|Type   |    Default Value    |
+|       -:|:-     |        :-:          |
+|      min|`int`  | `Integer.MIN_VALUE` |
+|      max|`int`  | `Integer.MAX_VALUE` |
+
+### Example
+```java
+@RangeInt(min = 0) 
+public static int thingFECapped = 50;
+```
+This will produce the following config:
+
+```
+# Min: 0
+# Max: 2147483647
+I:thingFECapped=50
+```
+
+@RangeDouble Use
+----------------
+
+You should use `@RangeDouble` to limit a `double` or `Double` config value. 
+
+It has a couple of properties:
+
+| Property|Type     |    Default Value    
+|       -:|:-       |        :-:         |
+|      min|`double` | `Double.MIN_VALUE` |
+|      max|`double` | `Double.MAX_VALUE` |
+
+### Example
+```java
+@RangeDouble(min = 0, max = Math.PI) 
+public static double chanceToDrop = 2;
+```
+This will produce the following config:
+```
+# Min: 0.0
+# Max: 3.141592653589793
+D:chanceToDrop=2.0
+```
+
+!!! note
+  There is not currenty (in 1.12.2) a `@RangedFloat` or `@RangedLong` or any other variant of the Ranged values. 
+
+@LangKey Use
+------------
+
+If you want to add translations for your configs in the mod options menu, add to the config's field `@LangKey`.
+
+This has 1 property:
+| Property | Type    | Default Value |
+|        -:|:-       |      :-:      |
+|     value|`String` |      N/A      | 
+
+@RequiresMcRestart Use
+----------------------
+
+You should include this annotation on a config that if changed, will require the game to be restarted for the change to take affect.
+
+### Example
+```java
+@RequiresMcRestart
+public static boolean overlayEnabled = false;
+```
+This will make the game require a restart if the value is changed in the configs menu.
+
+@RequiresWorldRestart Use
+-------------------------
+
+This will force the game to be restarted if the config is changed in the mod options menu.
+
+### Example
+```java
+@RequiresWorldRestart
+public static boolean someOtherworldlyThing = false;
+```
+This will force the world to be restarted if the config is changed in the mod options menu.
+
+Sub Categories
+--------------
+If you want Sub Categories, you must specify them. A good way to do this is to make a private inner class, and create a singleton instance in it's parent. 
```

Comment:
>If you want Sub Categories, you must specify them.

This could probably be removed. If you want *anything* you must specify it. No need to state that here.